pandas/core/generic.py

```diff
@@ -6112,6 +6112,14 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             raise NotImplementedError("Interpolation with NaNs in the index "
                                       "has not been implemented. Try filling "
                                       "those NaNs before interpolating.")
+        is_datetime = False
```

Comment:
this needs to be pushed down to the block in ``pandas/core/internals``, should go with ``def _interpolate`` I think

Reply:
I have to explore more about it. Will do that in a day or two.